### ISSUE TITLE
Not caching some requests

### ISSUE THREAD

**Issue by mhluska** [Appchance/ember-cached-shoe#4]
I'm noticing that some requests don't get cached. It works in many simple cases with a top level route but I have a scenario where this doesn't seem to work:

1. FastBoot runs and `ember-cached-shoe` caches all requests correctly.
2. The browser finishes loading the JS and shoebox <script> tags are correctly available in the DOM
3. _Something_ happens, maybe a re-render by Ember that causes the script tags to disappear from the body
4. Ember tries to make an AJAX call for some async data (maybe a has-many relationship or related data) but the shoebox script tags are now missing so the full request happens again, causing a flicker on the screen.

Have you experienced this issue? Any common workarounds? It sounds like a hack but if the shoebox data could be stored in cookies instead of DOM, it wouldn't get blown away in this scenario.

**Comment by mazondo** [comment]
We have this issue in production as well, we ended up making our own instance initializer to store it before fastboot deletes the shoebox.  the issue seems to be here:

https://github.com/ember-fastboot/ember-cli-fastboot/pull/569/files#diff-fa71c9edd502aa21b576d69c24dd4c63R31

We'd chatted about ways to solve it upstream, as it feels like a fastboot issue (shoebox isn't available after initial render).  original work there was done by @cibernox so perhaps he knows more?

**Comment by mhluska** [comment]
@mazondo I ended up just using [ember-data-storefront](https://github.com/embermap/ember-data-storefront) after a recommendation in Slack and it works perfectly out of the box.

**Comment by joshkg** [comment]
I can second @mhluska, was experiencing the same issue (which was really annoying me, as ember-cached-shoe has been working beautifully on all other routes) and swtiching to [ember-data-storefront](https://github.com/embermap/ember-data-storefront) solved it.

**Comment by mcfiredrill** [comment]
Yes for me its only working for simple routes using findAll, not other routes using a RSVP.hash. I could investigate more but I suppose I will give ember-data-storefront a try.
